inspirehep/modules/literaturesuggest/tasks.py

```diff
@@ -24,205 +24,249 @@
 
 import copy
 
-from datetime import date
 
-from flask import current_app
 
-from sqlalchemy.orm.exc import NoResultFound
+from invenio_accounts.models import User
+
 
 from idutils import is_arxiv_post_2007
-from invenio_accounts.models import User
-from invenio_oauthclient.models import UserIdentity
 from inspirehep.modules.forms.utils import filter_empty_elements
-from inspirehep.modules.search.api import JournalsSearch
 from inspirehep.utils.record import get_title, get_value
 
-from .dojson.model import literature
+from inspire_schemas.api import LiteratureBuilder
+
+from inspirehep.utils.pubnote import split_page_artid
+
+from inspirehep.utils.helpers import force_force_list
+from inspirehep.utils.normalizers import (
+    normalized_title,
+    normalized_orcid,
+)
 
 
 def formdata_to_model(obj, formdata):
     """Manipulate form data to match literature data model."""
+    def _filter_arxiv_categories(categories, arxiv_id):
+        arxiv_categories = categories.split()
+        if len(arxiv_id.split('/')) == 2:
+            if arxiv_id.split('/')[0] not in arxiv_categories:
+                arxiv_categories.append(arxiv_id.split('/')[0])
```

Comment:
* This will fail to validate if the arXiv category has been renamed, e.g. [funct-an/9707009](https://arxiv.org/abs/funct-an/9707009) has category `math.FA` now.
* The order of categories is meaningful, the primary one (the one in the arXiv ID) should always be the first.